packages/editor-ui/src/types/externalHooks.ts

```diff
@@ -0,0 +1,294 @@
+import type { N8nInput } from 'n8n-design-system';
+import type {
+	IConnections,
+	INodeParameters,
+	INodeProperties,
+	INodeTypeDescription,
+	ITelemetryTrackProperties,
+	INodeParameterResourceLocator,
+	NodeParameterValue,
+	ResourceMapperValue,
+} from 'n8n-workflow';
+import type { RouteLocation } from 'vue-router';
+import type {
+	AuthenticationModalEventData,
+	ExecutionFinishedEventData,
+	ExecutionStartedEventData,
+	ExpressionEditorEventsData,
+	InsertedItemFromExpEditorEventData,
+	NodeRemovedEventData,
+	NodeTypeChangedEventData,
+	OutputModeChangedEventData,
+	UpdatedWorkflowSettingsEventData,
+	UserSavedCredentialsEventData,
+} from '@/hooks';
+import type {
+	INodeCreateElement,
+	INodeUi,
+	INodeUpdatePropertiesInformation,
+	IPersonalizationLatestVersion,
+	IWorkflowDb,
+	NodeFilterType,
+} from '@/Interface';
+import type { ComponentPublicInstance } from 'vue/dist/vue';
+import type { useWebhooksStore } from '@/stores';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface ExternalHooksMethod<T = any, R = void> {
+	(store: ReturnType<typeof useWebhooksStore>, metadata: T): R | Promise<R>;
+}
+
+export interface ExternalHooksGenericContext {
+	[key: string]: ExternalHooksMethod[];
+}
+
+export interface ExternalHooks {
```

Comment:
would be great if we can maybe add a comment here not to have people add to this.. to avoid features expanding on this.